nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -0,0 +1,718 @@
+									"status": {
+										Type:        schema.TypeString,
+										Description: "Upgrade status",
+										Computed:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUpgradeGroupPlanSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Upgrade plan for this upgrade",
+		Optional:    true,
+		Computed:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:        schema.TypeString,
+					Description: "Component type",
+					Computed:    true,
+				},
+				"id": {
+					Type:        schema.TypeString,
+					Description: "ID of upgrade unit group",
+					Computed:    true,
+				},
+				"enabled": {
+					Type:        schema.TypeBool,
+					Description: "Flag to indicate whether upgrade of this group is enabled or not",
+					Computed:    true,
+				},
+				"parallel": {
+					Type:        schema.TypeBool,
+					Description: "Upgrade method to specify whether the upgrade is to be performed in parallel or serially",
+					Computed:    true,
+				},
+				"pause_after_each_upgrade_unit": {
+					Type:        schema.TypeBool,
+					Description: "Flag to indicate whether upgrade should be paused after upgrade of each upgrade-unit",
+					Computed:    true,
+				},
+				"extended_config": {
+					Type:     schema.TypeMap,
+					Optional: true,
+					Computed: true,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUpgradeGroupSchema(isHostGroup bool) *schema.Schema {
+	elemSchema := map[string]*schema.Schema{
+		"id": {
+			Type:        schema.TypeString,
+			Description: "ID of upgrade unit group",
+			Required:    true,
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Description: "Flag to indicate whether upgrade of this group is enabled or not",
+			Optional:    true,
+			Default:     true,
+		},
+		"parallel": {
+			Type:        schema.TypeBool,
+			Description: "Upgrade method to specify whether the upgrade is to be performed in parallel or serially",
+			Optional:    true,
+			Default:     true,
+		},
+		"pause_after_each_upgrade_unit": {
+			Type:        schema.TypeBool,
+			Description: "Flag to indicate whether upgrade should be paused after upgrade of each upgrade-unit",
+			Optional:    true,
+			Default:     false,
+		},
+	}
+
+	if isHostGroup {
+		elemSchema["upgrade_mode"] = &schema.Schema{
+			Type:         schema.TypeString,
+			Description:  "Upgrade mode",
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(supportedUpgradeMode, false),
+		}
+		elemSchema["maintenance_mode_config_vsan_mode"] = &schema.Schema{
+			Type:         schema.TypeString,
+			Description:  "Maintenance mode config vsan mode",
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(supportedMaintenanceModeConfigVsanMode, false),
+		}
+		elemSchema["maintenance_mode_config_evacuate_powered_off_vms"] = &schema.Schema{
```

Comment:
I think we should have a good reason to replace bool value with `true`, `false`, `unspecified`. In this case it seems that there should be a default..

Reply:
Ah sorry, I forget to change this one while I do the same thing for `rebootless_upgrade`. Done.